experimental/benchmark/src/backend.rs

```diff
@@ -0,0 +1,131 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use crate::{
+    database::Database,
+    proto::{
+        trusted_database_client::TrustedDatabaseClient,
+        trusted_database_server::{TrustedDatabase, TrustedDatabaseServer},
+        GetPointOfInterestRequest, GetPointOfInterestResponse, ListPointsOfInterestRequest,
+        ListPointsOfInterestResponse, PointOfInterest,
+    },
+};
+use log::{debug, error, info};
+use oak_abi::label::Label;
+use prost::Message;
+use std::collections::HashMap;
+use tokio::task::JoinHandle;
+use tonic::{
+    metadata::MetadataValue,
+    transport::{Channel, Server},
+    Request, Response, Status,
+};
+
+pub struct TrustedDatabaseService {
+    points_of_interest: HashMap<String, PointOfInterest>,
+}
+
+#[tonic::async_trait]
+impl TrustedDatabase for TrustedDatabaseService {
+    // Find Point Of Interest based on id.
+    async fn get_point_of_interest(
+        &self,
+        request: Request<GetPointOfInterestRequest>,
+    ) -> Result<Response<GetPointOfInterestResponse>, Status> {
+        debug!("Received request: {:?}", request);
+        match self.points_of_interest.get(&request.get_ref().id) {
+            Some(point) => {
+                debug!("Found Point Of Interest: {:?}", point);
+                Ok(Response::new(GetPointOfInterestResponse {
+                    point_of_interest: Some(point.clone()),
+                }))
+            }
+            None => {
+                let err = tonic::Status::new(tonic::Code::Internal, "Empty database");
+                error!("{:?}", err);
+                Err(err)
+            }
+        }
+    }
+
+    async fn list_points_of_interest(
+        &self,
+        _request: Request<ListPointsOfInterestRequest>,
+    ) -> Result<Response<ListPointsOfInterestResponse>, Status> {
+        unimplemented!();
+    }
+}
+
+pub struct TestBackend {
```

Comment:
nit: I think something like `TonicBackend` or `VanillaBackend` is more expressive. Similarly, I think `OakBackend` is a better name for `TestRuntime`.

You could also have a `Backend` trait with a `send_request` function, and two implementations for it, one with Tonic and one with Oak. But this is super-nit :) 

Perhaps also add a brief comment here explaining what this struct does.